RecoBTag/TensorFlow/plugins/DeepDoubleXTFJetTagsProducer.cc

```diff
@@ -248,27 +253,28 @@ void DeepDoubleXTFJetTagsProducer::produce(edm::Event& iEvent, const edm::EventS
   for (std::size_t i=0; i < lp_tensors_.size(); i++) {
     input_tensors[input_sizes.size() + i] = tensorflow::NamedTensor(lp_names_[i], lp_tensors_[i]);
   }
-
-  std::size_t n_batches = n_jets/n_batch_jets; // either 1 or n_jets
+  
+  std::size_t n_batches = n_jets_all/n_batch_jets; // either 1 or n_jets
   for (std::size_t batch_n=0; batch_n < n_batches; batch_n++) {
-
     // tensors have to be zeroed before filling per batch
     for (std::size_t i=0; i < input_sizes.size(); i++) {
       input_tensors[i].second.flat<float>().setZero();
     }
-
+    // don't run batch unless filled
+    bool run_this_batch = false;
     // fill values of the input tensors
     for (std::size_t jet_bn=0; jet_bn < (std::size_t) n_batch_jets; jet_bn++) {
```

Comment:
doesn't this solution completely break batched inference?
Note that with batching enabled ```n_batches = n_jets_all/n_batch_jets \equiv n_jets_all/n_jets```, which can even easily be ```n_jets_all / 0 ```

I think that keeping ```n_batches = n_jets/n_batch_jets``` should work, assuming that you can consistently skip the same jets as the ones used to define ```n_jets```.

Please also add a protection for n_jets == 0 in the math used to derive the n_batches

Reply:
Hmm, you are right. Keeping `n_batches = n_jets/n_batch_jets` works, but doesn't let us assign the -1 default in the producer, the need to be a default in the tensor filler.  I made a mistake of not updating the batch_eval line `n_batch_jets = batch_eval_ ? n_jets_all : 1;` This should restore the original logic imo.